packages/rum-core/src/domain/tracing/identifier.ts

```diff
@@ -0,0 +1,85 @@
+import { ExperimentalFeature, isExperimentalFeatureEnabled } from '@datadog/browser-core'
+import { getCrypto } from '../../browser/crypto'
+
+interface BaseIdentifier {
+  toString(radix?: number): string
+}
+
+export interface TraceIdentifier extends BaseIdentifier {
+  // We use a brand to distinguish between TraceIdentifier and SpanIdentifier, else TypeScript
+  // considers them as the same type
+  __brand: 'traceIdentifier'
+}
+
+export interface SpanIdentifier extends BaseIdentifier {
+  __brand: 'spanIdentifier'
+}
+
+export function createTraceIdentifier() {
+  return createIdentifier(64) as TraceIdentifier
+}
+
+export function createSpanIdentifier() {
+  return createIdentifier(63) as SpanIdentifier
+}
+
+let createIdentifierImplementationCache: ((bits: 63 | 64) => BaseIdentifier) | undefined
+
+function createIdentifier(bits: 63 | 64): BaseIdentifier {
+  if (!createIdentifierImplementationCache) {
+    createIdentifierImplementationCache =
+      isExperimentalFeatureEnabled(ExperimentalFeature.CONSISTENT_TRACE_SAMPLING) && areBigIntIdentifiersSupported()
+        ? createIdentifierUsingBigInt
+        : createIdentifierUsingUint32Array
+  }
+  return createIdentifierImplementationCache(bits)
+}
+
+export function areBigIntIdentifiersSupported() {
+  try {
+    crypto.getRandomValues(new BigUint64Array(1))
+    return true
+  } catch {
+    return false
+  }
+}
+
+function createIdentifierUsingBigInt(bits: 63 | 64): BaseIdentifier {
+  let id = crypto.getRandomValues(new BigUint64Array(1))[0]
+  if (bits === 63) {
+    // eslint-disable-next-line no-bitwise
+    id >>= BigInt('1')
+  }
+  return id
+}
+
+// TODO: remove this when all browser we support have BigInt support
+function createIdentifierUsingUint32Array(bits: 63 | 64): BaseIdentifier {
```

Comment:
👏 praise: Will be removed but this implementation with Uint Array is much cleaner.